src/Pos/ExecMode.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies        #-}
+
+module Pos.ExecMode
+    ( ExecMode(..)
+    , ExecModeM
+    ) where
+
+import           Universum
+
+import           Control.Monad.Base          (MonadBase)
+import           Control.Monad.Fix           (MonadFix)
+import qualified Control.Monad.Reader        as Mtl
+import           Control.Monad.Trans.Control (MonadBaseControl (..))
+import           Data.Coerce                 (coerce)
+import qualified Ether
+import           Ether.Internal              (HasLens (..))
+import           Mockable                    (ChannelT, Counter, Distribution, Gauge,
+                                              MFunctor' (..), Mockable (..), Promise,
+                                              SharedAtomicT, SharedExclusiveT, ThreadId)
+import           System.Wlog                 (CanLog)
+
+import           Pos.Util.Util               (PowerLift (..))
+
+type family ExecModeM mode :: * -> *
+
+newtype ExecMode mode a = ExecMode { unExecMode :: ExecModeM mode a }
```

Comment:
The `mode` parameter to this type determines the instances. For each mode there's a tag:

```
data REAL ssc -- real mode
data LW -- light wallet
data WWEB -- wallet web mode
data WEB -- web mode
```

So, the way to think about this type is that it's a base redirection layer. The alternative would be to use a `TaggedTrans` here, but this newtype indeed hides the type to decrease resource consumption.